cogs/displayinfo.py

```diff
@@ -33,85 +33,113 @@ class DisplayInfo(commands.Cog):
                         if currentStartTime:
                             playTime = datetime.now() - currentStartTime # Calculate the time difference (playTime) between the start and end time of the activity
-                            self.timers[mem_before.name][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[mem_before.name][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
+                            self.timers[before_string_id][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[before_string_id][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
                         print("end:", self.timers) # printing the dictionary to see the updates after an activity is ended
                         self.current_collection.replace_one({}, self.timers) # Update the MongoDB collection with the updated activity data(self.timers)
 
-    # This function is an asynchronous coroutine that runs indefinitely in the background when mem_after.activity is True.
-    # It is designed to update activity time information for a Discord member ('mem_after') whenever they start a new activity every 5 minutes.
-    async def timed_update(self, mem_before: discord.Member, mem_after: discord.Member):
-        await asyncio.sleep(300)
-        while not mem_before.activity and mem_after.activity:
-            playTime = 300 # add 5 mins to activity time
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["totalTime"] = self.timers[mem_after.name][self.current_date][mem_after.activity.name].get("totalTime", 0) + playTime
-            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["startTime"] = datetime.now()
-            print("in timed_update", self.timers) #printing the dictionary to see the timed updates
+    async def update_now(self, string_id):
+        if string_id in self.timers and self.current_date in self.timers[string_id]:
+            for activity, info in self.timers[string_id][self.current_date].items():
+                if info["startTime"] != None:
+                     if activity in self.timers[string_id][self.current_date]:
+                        currentStartTime = self.timers[string_id][self.current_date][activity]["startTime"]
+                        playTime = datetime.now() - currentStartTime 
+                        self.timers[string_id][self.current_date][activity]["totalTime"] = self.timers[string_id][self.current_date][activity].get("totalTime", 0) + playTime.total_seconds()
+                        self.timers[string_id][self.current_date][activity]["startTime"] = datetime.now()
+
+    async def timed_update(self):
+        while len(self.members_online) != 0:
```

Comment:
NIT: Shouldn't be needed if we are starting/canceling `timed_update()` appropriately.

Reply:
To be safer if we want to leave it here we should do `> 0`